test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractDeleteSuite.java

```diff
@@ -138,6 +155,103 @@ private HapiApiSpec deleteTransfersToContract() {
+		return defaultHapiSpec("DeleteWithZeroBalanceDoesNotTransferToAccount")
+				.given(
+						fileCreate("contractBytecode").path(PAYABLE_CONSTRUCTOR),
+						cryptoCreate("receiver").balance(0L),
+						contractCreate("toBeDeleted").bytecode("contractBytecode").balance(0L)
+				).when(
+						contractDelete("toBeDeleted").transferAccount("receiver")
+				).then(
+						getAccountBalance("receiver").hasTinyBars(0L)
+				);
+	}
+
+	private HapiApiSpec deleteWithZeroBalanceDoesNotTransferToContract() {
+		return defaultHapiSpec("DeleteWithZeroBalanceDoesNotTransferToContract")
+				.given(
+						fileCreate("contractBytecode").path(PAYABLE_CONSTRUCTOR),
+						contractCreate("receiver").balance(0L),
+						contractCreate("toBeDeleted").bytecode("contractBytecode").balance(0L)
+				).when(
+						contractDelete("toBeDeleted").transferContract("receiver")
+				).then(
+						getAccountBalance("receiver").hasTinyBars(0L)
+				);
+	}
+
+	private HapiApiSpec deleteFailsWithObtainerAccountSameContractID() {
+		return defaultHapiSpec("DeleteFailsWithObtainerAccountSameContractID")
+				.given(
+						fileCreate("contractBytecode").path(PAYABLE_CONSTRUCTOR),
+						contractCreate("toBeDeleted").bytecode("contractBytecode").balance(1L)
+				)
+				.when(
+						contractDelete("toBeDeleted").transferAccount("toBeDeleted").hasKnownStatus(OBTAINER_SAME_CONTRACT_ID)
+				)
+				.then(
+						getAccountBalance("toBeDeleted").hasTinyBars(1L)
+				);
+	}
+
+	private HapiApiSpec deleteFailsWithObtainerContractSameContractID() {
+		return defaultHapiSpec("DeleteFailsWithObtainerContractSameContractID")
+				.given(
+						fileCreate("contractBytecode").path(PAYABLE_CONSTRUCTOR),
+						contractCreate("toBeDeleted").bytecode("contractBytecode").balance(1L)
+				)
+				.when(
+						contractDelete("toBeDeleted").transferContract("toBeDeleted").hasKnownStatus(OBTAINER_SAME_CONTRACT_ID)
+				)
+				.then(
+						getAccountBalance("toBeDeleted").hasTinyBars(1L)
+				);
+	}
+
+	private HapiApiSpec deleteFailsWithObtainerDoesNotExist() {
+		HapiSpecSetup.getDefaultInstance().invalidContract();
+		return defaultHapiSpec("DeleteFailsWithObtainerDoesNotExist")
+				.given(
+						fileCreate("contractBytecode").path(PAYABLE_CONSTRUCTOR),
+						cryptoCreate("receiver").balance(0L),
+						contractCreate("toBeDeleted").bytecode("contractBytecode").balance(1L)
+				)
+				.when(
+						cryptoDelete("receiver"),
+						contractDelete("toBeDeleted").transferAccount("receiver").hasKnownStatus(OBTAINER_DOES_NOT_EXIST)
+				)
+				.then(
+						getAccountBalance("toBeDeleted").hasTinyBars(1L)
+				);
+	}
+
+	private HapiApiSpec deleteFailsWithAccountExpiredAndPendingRemoval() {
+		return defaultHapiSpec("DeleteFailsWithAccountExpiredAndPendingRemoval")
+				.given(
+						fileUpdate(APP_PROPERTIES)
+								.payingWith(ADDRESS_BOOK_CONTROL)
+								.overridingProps(
+										Map.of("autorenew.isEnabled", "true",
+												"ledger.autoRenewPeriod.minDuration", "" + 1)),
```

Comment:
I think that you won't be able to execute this case (with the desired effect) with `autorenew.isEnabled=true`

Reply:
The test case passes with this setting